### PR TITLE
pgcraft sampling using hwmap

### PR DESCRIPTION
**Pull request by cboesch-gpsw**

### CODE REVIEW

File: libnodegl/hwmap.c [hunk 1]
```diff
@@ -125,18 +125,46 @@ static int exec_hwconv(struct hwmap *hwmap)
     return 0;
 }
 
+static const struct hwmap_class **get_extra_hwmap_classes(int backend)
```
Comment (mbouron-gpsw) [inline libnodegl/hwmap.c:128]: `get_backend_hwmap_classes` ?

**Comment by mbouron-gpsw** [comment]
` hwmap: expose a function to inspect available accelerations`, maybe rename to ` hwmap: expose a function to inspect available image layouts` as  a `NGLI_IMAGE_LAYOUT_{DEFAULT,NV12,YUV}` can be used in both cases (cpu / hw-interop).

File: libnodegl/hwmap.c (hunk 1 above)
Comment (cboesch-gpsw) [reply libnodegl/hwmap.c]: renamed

**Comment by cboesch-gpsw** [comment]
> ` hwmap: expose a function to inspect available accelerations`, maybe rename to ` hwmap: expose a function to inspect available image layouts` as a `NGLI_IMAGE_LAYOUT_{DEFAULT,NV12,YUV}` can be used in both cases (cpu / hw-interop).

Done:
- `hwmap: expose a function to inspect available image layouts`
- `pgcraft: use hwmap to inspect available image layouts`

**Review by mbouron-gpsw — APPROVED** [review]

**Pull request merged**